Userland/Libraries/LibJS/Runtime/Map.cpp

```diff
@@ -22,6 +22,60 @@ Map::~Map()
 {
 }
 
+void Map::map_clear()
+{
+    m_keys.clear();
+    m_entries.clear();
+}
+
+bool Map::map_remove(Value const& key)
+{
+    Optional<size_t> index;
+
+    for (auto it = m_keys.begin(); it != m_keys.end(); ++it) {
+        if (it->operator==(key)) {
```

Comment:
This should not be `==` which calls `same_value` but `same_value_zero` and the same holds for all comparisons.

This might be possible with a special `Trait` but you probably know more about that than me.

Reply:
I'll just call into `ValueTraits::equals`.